src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotSourceDatasetDependencyDataGcpStep.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.common.FlightUtils;
+import bio.terra.common.exception.PdaoException;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.service.dataset.flight.ingest.OptionalStep;
+import bio.terra.service.filedata.google.firestore.FireStoreDependencyDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.google.cloud.bigquery.BigQueryException;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+public class DeleteSnapshotSourceDatasetDependencyDataGcpStep extends OptionalStep {
+  private FireStoreDependencyDao dependencyDao;
+  private UUID snapshotId;
+  private DatasetService datasetService;
+
+  public DeleteSnapshotSourceDatasetDependencyDataGcpStep(
+      FireStoreDependencyDao dependencyDao,
+      UUID snapshotId,
+      DatasetService datasetService,
+      Predicate<FlightContext> doCondition) {
+    super(doCondition);
+    this.dependencyDao = dependencyDao;
+    this.snapshotId = snapshotId;
+    this.datasetService = datasetService;
+  }
+
+  @Override
+  public StepResult doOptionalStep(FlightContext context) throws InterruptedException {
+    FlightMap map = context.getWorkingMap();
+    UUID datasetId = map.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
+    try {
+      Dataset dataset = datasetService.retrieve(datasetId);
+      dependencyDao.deleteSnapshotFileDependencies(dataset, snapshotId.toString());
+
+    } catch (BigQueryException ex) {
```

Comment:
When would this step throw a `BigQueryException`? I thought it just cleaned up firestore records

Reply:
Good point - that's a bad copy from when I split this into two steps for GCP.